lib/cache.go

```diff
@@ -0,0 +1,57 @@
+package lib
+
+import (
+	"net/http"
+	"time"
+)
+
+type CacheEntry struct {
+	Data      []byte
+	CreatedAt time.Time
+	ExpiresIn *time.Duration
+	Headers   http.Header
+}
+
+func (c *CacheEntry) Expired() bool {
+	if c.ExpiresIn == nil {
+		return false
```

Comment:
I would rather return true here, having a cache entry that never expires sounds like something we want to avoid at all costs